src/Components/Analyzers/src/ComponentParameterUsageAnalyzer.cs

```diff
@@ -0,0 +1,98 @@
+                    return;
+                }
+
+                context.RegisterOperationBlockStartAction(startBlockContext =>
+                {
+                    var assignedPropertyReferences = new List<IPropertyReferenceOperation>();
+                    startBlockContext.RegisterOperationAction(context =>
+                    {
+                        foreach (var operation in context.Operation.Children)
+                        {
+                            if (operation.Kind == OperationKind.PropertyReference)
+                            {
+                                assignedPropertyReferences.Add((IPropertyReferenceOperation)operation);
+                            }
+                        }
+                    }, OperationKind.SimpleAssignment, OperationKind.CompoundAssignment, OperationKind.CoalesceAssignment);
+
+                    startBlockContext.RegisterOperationBlockEndAction(context =>
+                    {
+                        for (var i = 0; i < assignedPropertyReferences.Count; i++)
+                        {
+                            var propertyReference = assignedPropertyReferences[i];
+                            var componentProperty = propertyReference.Member;
+                            var propertyAttributes = componentProperty.GetAttributes();
+
+                            if (!ComponentFacts.HasParameterAttribute(symbols, propertyAttributes))
+                            {
+                                // This is not a property reference that we care about, it is not decorated with [Parameter].
+                                continue;
+                            }
+
+                            var propertyLocationContainingType = componentProperty.ContainingType;
+                            if (!ComponentFacts.IsComponent(symbols, propertyLocationContainingType))
+                            {
+                                // Someone referenced a property as [Parameter] inside something that is not a component.
+                                continue;
+                            }
+
+                            var assignmentLocationContainingType = startBlockContext.OwningSymbol?.ContainingType;
+                            if (assignmentLocationContainingType == null)
+                            {
+                                // Assignment location has no containing type. Most likely we're operating on malformed code, don't try and validate.
+                                continue;
+                            }
+
+                            if (ComponentFacts.IsAssignableFrom(propertyLocationContainingType, assignmentLocationContainingType))
+                            {
+                                // The assignment is taking place within the components type hierarchy. This means the user is setting this in a supported
+                                // scenario.
+                                continue;
+                            }
```

Comment:
I'm hoping that we don't need this as a utility function that we have to maintain. That doesn't seem right. Look at `ClassifyConversion`, I have a usage somewhere.

Reply:
Interesting, didn't know it existed. Will give it a shot